### PR TITLE
Teacher dashboard

### PR DESCRIPTION
**Pull request by jsparvath**
Added methods enabling seeing the users reading_session activity. This allows us to see the user activity even if there are no bookmarks related to the session

### CODE REVIEW

File: zeeguu_api/api/teacher_dashboard.py [hunk 1]
```diff
@@ -272,6 +271,20 @@ def _link_teacher_cohort(user_id, cohort_id):
...
+    '''
+        Return list of dictionaries containing user info for all cohorts that the logged in user owns.
+
+    '''
+    from zeeguu_core.model import TeacherCohortMap
+    mappings = TeacherCohortMap.query.filter_by(user_id=flask.g.user.id).all()
+    all_users = []
+    for m in mappings:
+        users = users_from_cohort(m.cohort_id, duration)
+        all_users.extend(json.loads(users))
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: i'm a bit puzzled by this. 

are you sure that converting to json and then deconverting 
again here is the nicest way to solve the problem you have?
i wonder if you could refactor the other function to extract a local 
that you can reuse both here and there ... something like: 

```
user_info_from_cohort
        c = Cohort.query.filter_by(id=id).one()
        users = User.query.filter_by(cohort_id=c.id).all()
        users_info = []
        for u in users:
            info = _get_user_info_for_teacher_dashboard(u.id, duration)
            users_info.append(info)
       return users_info
```

File: zeeguu_api/api/teacher_dashboard.py [hunk 2]
```diff
@@ -343,6 +356,28 @@ def add_teacher(id):
...
+        flask.abort(400)
+        return "NoUserFound"
+
+    if (not has_permission_for_cohort(user.cohort_id)):
+        flask.abort(401)
+
+    now = datetime.today()
+    date = now - timedelta(days=int(time_period));
+
+    # True input causes function to return context too.
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: i can't parse this comment. what does it mean?

Comment (jsparvath) [reply zeeguu_api/api/teacher_dashboard.py]: Ah, it was a leftover from the method I borrowed. removed it

File: zeeguu_api/api/teacher_dashboard.py (hunk 1 above)
Comment (jsparvath) [reply zeeguu_api/api/teacher_dashboard.py]: Good point. much cleaner now :)

**Pull request merged**